### PR TITLE
feat: BREAKING CHANGE add equality, hashcode and toString helper overrides to autogenerated code

### PR DESCRIPTION
**Pull request by td-famedly**
autogen script change: https://github.com/famedly/dart_openapi_codegen/pull/50

### CODE REVIEW

File: lib/matrix_api_lite/generated/model.dart [hunk 1]
```diff
@@ -82,6 +106,20 @@ class DiscoveryInformation {
...
   Map<String, Map<String, Object?>> additionalProperties;
+
+  @override
+  bool operator ==(Object other) =>
+      other is DiscoveryInformation &&
+      other.mHomeserver == mHomeserver &&
+      other.mIdentityServer == mIdentityServer;
+
+  @override
+  int get hashCode => mHomeserver.hashCode ^ mIdentityServer.hashCode;
```
Comment (nico-famedly) [inline lib/matrix_api_lite/generated/model.dart:117]: Pretty sure this should be

```suggestion
  int get hashCode => Object.hash(mHomeserver, mIdentityServer);
```

Comment (td-famedly) [reply lib/matrix_api_lite/generated/model.dart]: done

File: lib/matrix_api_lite/generated/model.dart [hunk 2]
```diff
@@ -209,6 +276,20 @@ class GenerateLoginTokenResponse {
 
   /// The login token for the `m.login.token` login flow.
   String loginToken;
+
+  @override
+  bool operator ==(Object other) =>
+      other is GenerateLoginTokenResponse &&
+      other.expiresInMs == expiresInMs &&
+      other.loginToken == loginToken;
```
Comment (nico-famedly) [inline lib/matrix_api_lite/generated/model.dart:284]: This should probably not be in the logs by default?

File: lib/matrix_api_lite/generated/model.dart [hunk 3]
```diff
@@ -834,6 +1118,22 @@ class ThreePidCredentials {
 
   /// The session identifier given by the identity server.
   String sid;
+
+  @override
+  bool operator ==(Object other) =>
+      other is ThreePidCredentials &&
+      other.clientSecret == clientSecret &&
+      other.idAccessToken == idAccessToken &&
```
Comment (nico-famedly) [inline lib/matrix_api_lite/generated/model.dart:1126]: Should we really log this via toString by default?

File: lib/matrix_api_lite/generated/model.dart [hunk 4]
```diff
@@ -1703,6 +2291,26 @@ class LoginResponse {
   /// optionally validating the URLs within. This object takes the same
   /// form as the one returned from .well-known autodiscovery.
   DiscoveryInformation? wellKnown;
+
+  @override
+  bool operator ==(Object other) =>
+      other is LoginResponse &&
+      other.accessToken == accessToken &&
```
Comment (nico-famedly) [inline lib/matrix_api_lite/generated/model.dart:2298]: We probably shouldn't log this via toString by default either.

**Review by nico-famedly — COMMENTED** [review]
I'm not sure about the toString stuff. We just had an access token leak and we also now might log user data more easily. I think we really should have to opt in to log stuff like that by default?

File: lib/matrix_api_lite/generated/model.dart (hunk 4 above)
Comment (td-famedly) [reply lib/matrix_api_lite/generated/model.dart:2298]: removed toString

**Review by nico-famedly — APPROVED** [review]

**Pull request merged**